File: src/test/java/com/bship/games/models/GameTest.java
package com.bship.games.models;

import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import java.util.Arrays;
import java.util.List;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.not;
import static org.hamcrest.Matchers.sameInstance;

public class GameTest {

    @Rule
    public ExpectedException thrown = ExpectedException.none();
    private Game game;

    @Before
    public void setup() {
        List<Board> boards = Arrays.asList(Board.builder().build(), Board.builder().build());
        game = Game.builder().withId(19L).withBoards(boards).build();
    }

    @Test
    public void copy_shouldCopyTheGame() {
        Game gameCopy = game.copy().build();

        assertThat(game, is(equalTo(gameCopy)));
        assertThat(gameCopy, is(not(sameInstance(game))));
    }

    @Test
    public void boardsShouldBeAnImmutableList() {
        thrown.expect(UnsupportedOperationException.class);

        game.getBoards().add(Board.builder().build());
    }
}

File: game/src/test/java/com/bship/games/endpoints/cabinet/entity/GameTest.kt
package com.bship.games.endpoints.cabinet.entity

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.Before
import org.junit.Test
import java.util.*

class GameTest {

    private lateinit var game: Game

    @Before
    fun setup() {
        val boards = Arrays.asList(Board.build {}, Board.build {})
        game = Game.builder().withId(19L).withBoards(boards).build()
    }

    @Test
    fun copy_shouldCopyTheGame() {
        val gameCopy = game.copy()

        assertThat(game).isEqualToComparingFieldByField(gameCopy)
        assertThat(game).isNotSameAs(gameCopy)
    }

    @Test
    fun boardsShouldBeAnImmutableList() {
        assertThatThrownBy { game.boards.add(Board.build {}) }
                .isInstanceOf(UnsupportedOperationException::class.java)
    }
}